Code/Tools/AssetProcessor/native/utilities/PlatformConfiguration.cpp

```diff
@@ -1569,29 +1572,49 @@ namespace AssetProcessor
 
         auto* fileStateInterface = AZ::Interface<AssetProcessor::IFileStateRequests>::Get();
 
-        QDir cacheRoot;
-        AssetUtilities::ComputeProjectCacheRoot(cacheRoot);
-
+        QString intermediateAssetsAsQString; // lazily computed on first use
```

Comment:
This seems like it could just be inside the loop, it's only run once I believe since skip is set to false after the first iteration

Reply:
I'll move it

Reply:
so it can't be in the loop, but it can be computed up front, once.  The result (intermediateAssetsAsQstring) has to survive looping, because its not necessarily the case that the first scan folder is the intermediate assets folder, so it could use it over and over until it finds it.